### ISSUE TITLE
Allow the use of fixed default values in ${...} variable definitions

### ISSUE THREAD

**Issue by arabold** [serverless/serverless#3900]
# This is a Feature Proposal

## Description

**Expected behavior:** `${opt:myopt, "bar"}` and/or `${opt:myopt, bar}` should print the value of the `myopt` command line option if set or the string "bar" otherwise.

Right now this isn't possible AFAIK. Instead you have to work around by defining an intermediate variable like this:
```yaml
custom:
  bar_string: bar
  myopt: ${opt:myopt, $self:custom.bar_string}
```

## Additional Data

* ***Serverless Framework Version you're using***: 1.16

**Comment by HyperBrain** [comment]
It would be great to have this, but not only restricted to string values. #3891 implements special handling for variables, like setting them to **null** which leads to omission.

So having `myopt: ${opt:myopt, null}` should also be possible as well as something like `myopt: ${opt:myopt, false}` or `myopt: ${opt:myopt, { a: 'foo', b: 'bar' }}`.

I'm not really sure if the latter is needed, but I would at least vote for the first two additions.

**Comment by shatgupt** [comment]
Thank you for raising this. When I started working with Serverless, I expected it to work the way you described. I discovered after some head scratching that I am not doing wrong but it was designed to be like that.

**Comment by alexdebrie** [comment]
Will be fixed by #4097

**Comment by pmuens** [comment]
Fixed with #4097
